crates/rover-client/src/operations/subgraph/fetch_all/service.rs

```diff
@@ -92,9 +92,24 @@ where
             inner
                 .call(GraphQLRequest::<SubgraphFetchAllQuery>::new(variables))
                 .await
-                .map_err(|err| RoverClientError::Service {
-                    source: Box::new(err),
-                    endpoint_kind: EndpointKind::ApolloStudio,
+                //.map_err(|err| RoverClientError::Service {
+                    //source: Box::new(err),
+                    //endpoint_kind: EndpointKind::ApolloStudio,
+                //})
+                .map_err(|err| {
+                    match err {
+                        GraphQLServiceError::InvalidCredentials() => {
+                            RoverClientError::PermissionError {
+                                msg: "attempting to fetch subgraphs".to_string(),
+                            }
+                        }
+                        _ => {
+                            RoverClientError::Service {
+                                source: Box::new(err),
+                                endpoint_kind: EndpointKind::ApolloStudio,
+                            }
+                        }
+                    }
                 })
```

Comment:
I think this came out all right. This maps the GraphQLServiceError to a RoverClientError of the correct flavor. I didn't cause this PermissionError to track source, which seems like a good idea to implement (but later).